### ISSUE TITLE
2nd Super admin can not delete account and log error comes

### ISSUE THREAD

**Issue by Vaibhavagarwal0807** [SocialEngine/phpv4-issues#189]
Delete button is not shown for 1st super admin in both Admin and User Side, but as soon as there is 1 more super admin, the delete button is shown.

1) When I try to delete this super admin from Manage Members in admin panel, then the smooth box opens and collapse to very small window and I can not delete this super admin until I change the member level.

2) When the new super admin tries to delete his account from user end >> Settings section, then a log error comes:

```
exception 'User_Model_Exception' with message 'Cannot delete superadmins.' in /home/sesdemo/public_html/application/modules/User/Model/User.php:372
Stack trace:
#0 /home/sesdemo/public_html/application/libraries/Zend/Db/Table/Row/Abstract.php(622): User_Model_User->_delete()
#1 /home/sesdemo/public_html/application/modules/User/controllers/SettingsController.php(569): Zend_Db_Table_Row_Abstract->delete()
#2 /home/sesdemo/public_html/application/libraries/Zend/Controller/Action.php(516): User_SettingsController->deleteAction()
#3 /home/sesdemo/public_html/application/libraries/Zend/Controller/Dispatcher/Standard.php(308): Zend_Controller_Action->dispatch('deleteAction')
#4 /home/sesdemo/public_html/application/libraries/Zend/Controller/Front.php(954): Zend_Controller_Dispatcher_Standard->dispatch(Object(Zend_Controller_Request_Http), Object(Zend_Controller_Response_Http))
#5 /home/sesdemo/public_html/application/modules/Core/Bootstrap.php(83): Zend_Controller_Front->dispatch()
#6 /home/sesdemo/public_html/application/libraries/Engine/Application.php(160): Core_Bootstrap->run()
#7 /home/sesdemo/public_html/application/index.php(214): Engine_Application->run()
#8 /home/sesdemo/public_html/index.php(24): include('/home/sesdemo/p...')
#9 {main}
```

In this case, some nice message can be shown instead of error log.

**Comment by Anshuse** [comment]
Hi @Vaibhavagarwal0807 ,

1) Regarding "When I try to delete this super admin from Manage":
When we click on the "delete" link from the "Admin" > "Manage" > "Manage Members" then pop-up doesn't get opened and when we open this "delete" link in the new tab then an error message shows up there. The reason behind such behavior is because super-admin can't be deleted from the site but yes, we should display a message instead of an error and we're reporting this at our end to work on this in near future.

2) Regarding "When the new super admin tries to delete his account from user end >> Settings section,":
Yes, you're right, we should display a direct message something like "Super-admins" can't be deleted . We've reported this issue at our end for our team to take appropriate action on this.

Thanks!

**Comment by Elshara** [comment]
Sometimes this error will pop up for random members their accounts cannot be deleted. I propose a simpler approach. Allow super admins the ability to delete any user via a setting in member levels,but do not give inferior member levels the ability to delete people above their level.

**Comment by DonnaScriptTechs** [comment]
This is fixed in 4.9.0 when it is released.

**Comment by Elshara** [comment]
Awesome.
